lib/taskana-core-test/src/test/java/acceptance/task/create/CreateTaskWithAttachmentAccTest.java

```diff
@@ -0,0 +1,201 @@
+            Pair.of("Company of objRef is null", objRefCompanyNull));
+
+    ThrowingConsumer<Pair<String, ObjectReference>> test =
+        p -> {
+          Task task = taskService.newTask(defaultWorkbasketSummary.getId());
+          task.setClassificationKey(defaultClassificationSummary.getKey());
+          task.setPrimaryObjRef(defaultObjectReference);
+          Attachment attachment = taskService.newAttachment();
+          attachment.setClassificationSummary(defaultClassificationSummary);
+          attachment.setObjectReference(p.getRight());
+          task.addAttachment(attachment);
+          assertThatThrownBy(() -> taskService.createTask(task))
+              .isInstanceOf(InvalidArgumentException.class);
+        };
+
+    return DynamicTest.stream(valuesForTests.iterator(), Pair::getLeft, test);
+  }
+
+  @WithAccessId(user = "user-1-1")
+  @TestFactory
+  Stream<DynamicTest>
+      should_ThrowException_When_CreatingTaskWithInvalidClassificationOfAttachment() {
+    Classification nonExistingClassification =
+        classificationService.newClassification(
+            "123key345", defaultWorkbasketSummary.getDomain(), "non existing type");
+    Classification classificationWithoutKey =
+        classificationService.newClassification(
+            null, defaultClassificationSummary.getDomain(), defaultClassificationSummary.getType());
+    List<Quadruple<String, Classification, String, Class<?>>> valuesForTests =
+        List.of(
+            Quadruple.of(
+                "Classification is null",
+                null,
+                "Classification of Attachment must not be null",
+                InvalidArgumentException.class),
+            Quadruple.of(
+                "Classification doesn't exist",
+                nonExistingClassification,
+                "Classification with key '123key345' and domain '"
+                    + defaultWorkbasketSummary.getDomain()
+                    + "' could not be found",
+                ClassificationNotFoundException.class),
+            Quadruple.of(
+                "Classification has no key",
+                classificationWithoutKey,
+                "ClassificationKey of Attachment must not be empty.",
+                InvalidArgumentException.class));
+
+    ThrowingConsumer<Quadruple<String, Classification, String, Class<?>>> test =
+        q -> {
+          Task task = taskService.newTask(defaultWorkbasketSummary.getId());
+          task.setClassificationKey(defaultClassificationSummary.getKey());
+          task.setPrimaryObjRef(defaultObjectReference);
+          Attachment attachment = taskService.newAttachment();
+          attachment.setClassificationSummary(q.getSecond());
+          attachment.setObjectReference(defaultObjectReference);
+          task.addAttachment(attachment);
+          assertThatThrownBy(() -> taskService.createTask(task))
+              .isInstanceOf(q.getFourth())
+              .hasMessageContaining(q.getThird());
```

Comment:
separate using the given-when-then theorem

Reply:
rearrange, so that Attachment gets created before Task (so that all setters of the Task are in close proximity)

Reply:
create variables for each value of the quadruple.

Reply:
I will create the variables